### PR TITLE
generate: make interface consistent with sync and fmt

### LINKED ISSUE #334: generate: add `-e, --exercise` option

### LINKED ISSUE #542: generate: make interface consistent with `sync` and `fmt`
`sync` and `fmt` make no changes unless `--update` is passed.

`generate` should work the same way.

Follow-up from https://github.com/exercism/configlet/issues/465

### PR DESCRIPTION
**Pull request by ee7**
To-do:
- [x] Merge #618 
- [x] Merge #620
- [x] Make `generate.nim` non-destructive by default, and support `-u`, `-y` and `-e`
- [ ] Update integration tests for `generate`
- [x] Update completion scripts

Closes: #334
Closes: #542

### CODE REVIEW

File: src/cli.nim [hunk 1]
```diff
@@ -51,6 +51,10 @@ type
       exerciseFmt*: string
       updateFmt*: bool
       yesFmt*: bool
+    of actGenerate:
+      exerciseGenerate*: string
+      updateGenerate*: bool
+      yesGenerate*: bool
```
Comment (ee7) [inline src/cli.nim:62]: We still can't write just use the name `exercise` for every branch that needs it them. (as [Nim RFC 368](https://www.github.com/nim-lang/RFCs/issues/368) wasn't accepted).

File: src/cli.nim [hunk 2]
```diff
@@ -87,12 +91,12 @@ type
     # Options for `completion`
     optCompletionShell = "shell"
 
-    # Options for `create`, `fmt` and `sync`
-    optFmtSyncCreateExercise = "exercise"
+    # Options for `create`, `fmt`, generate, and `sync`
+    optFmtGenerateSyncCreateExercise = "exercise"
```
Comment (ee7) [inline src/cli.nim:104]: Yes, the option names are getting ridiculous here.

Possible follow-up PR: change to `optExercise`.

**Review by ErikSchierboom — APPROVED** [review]
Good idea

**Comment by ErikSchierboom** [comment]
Due to limited time, I've not added new tests but only fixed the existing ones.

**Pull request merged**